samples/go/json-import/json_importer.go

```diff
@@ -8,16 +8,68 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"log"
+	"math/big"
 	"net/http"
 	"os"
+	"time"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/spec"
 	"github.com/attic-labs/noms/go/util/jsontonoms"
+	"github.com/dustin/go-humanize"
 	flag "github.com/juju/gnuflag"
 )
 
+func newProgressReader(r io.Reader) io.Reader {
```

Comment:
Agreed re: sharing of code – sorry, didn't spend enough time getting familiar with the code base before diving in.

I'll rewriting with the more callbacky style. I started out doing something very similar before veering off in this other direction.